stumpy/aamp.py

```diff
@@ -97,42 +98,52 @@ def _compute_diagonal(
         else:
             iter_range = range(-k, min(n_A - m + 1, n_B - m + 1 - k))
 
-        for i in iter_range:
+        for signed_i in iter_range:
+            i = np.uint64(signed_i)
+            j = np.uint64(i + k)
+
             if i == 0 or (k < 0 and i == -k):
                 p_norm = (
-                    np.linalg.norm(T_B[i + k : i + k + m] - T_A[i : i + m], ord=p) ** p
+                    np.linalg.norm(
+                        T_B[j : j + unsigned_m] - T_A[i : i + unsigned_m], ord=p
+                    )
+                    ** p
                 )
             else:
+                prev_i = np.uint64(i - 1)
```

Comment:
Instead of using `prev_i` and `prev_j`, at the top of this function, I would declare `uint64_1 = np.uint64(1)`. Then:

```
p_norm = np.abs(
                    p_norm
                    - np.absolute(T_B[uint64_j - uint64_1] - T_A[uint64_i - uint64_1]) ** p
                    + np.absolute(T_B[uint64_j + uint64_m - uint64_1] - T_A[uint64_i + uint64_m - uint64_1])
                    ** p
                )
```